price-feeder/oracle/oracle.go

```diff
@@ -352,28 +361,36 @@ func (o *Oracle) getOrSetProvider(ctx context.Context, providerName string) (pro
 	return priceProvider, nil
 }
 
-// filterDeviations find the standard deviations of the prices of
-// all assets, and filter out any providers that are not within 2𝜎 of the mean.
-func (o *Oracle) filterDeviations(
-	prices map[string]map[string]provider.TickerPrice) (
-	map[string]map[string]provider.TickerPrice, error,
-) {
+// filterTickerDeviations finds the standard deviations of the prices of
+// all assets, and filters out any providers that are not within 2𝜎 of the mean.
+func (o *Oracle) filterTickerDeviations(
+	prices provider.AggregatedProviderPrices,
+) (provider.AggregatedProviderPrices, error) {
 	var (
-		filteredPrices = make(map[string]map[string]provider.TickerPrice)
-		threshold      = sdk.MustNewDecFromStr("2")
+		filteredPrices = make(provider.AggregatedProviderPrices)
+		priceMap       = make(map[string]map[string]sdk.Dec)
 	)
 
-	deviations, means, err := StandardDeviation(prices)
+	for providerName, providerPrices := range prices {
+		if _, ok := priceMap[providerName]; !ok {
+			priceMap[providerName] = make(map[string]sdk.Dec)
+		}
+		for base, price := range providerPrices {
+			priceMap[providerName][base] = price.Price
+		}
+	}
+
+	deviations, means, err := StandardDeviation(priceMap)
 	if err != nil {
-		return make(map[string]map[string]provider.TickerPrice), nil
+		return nil, err
 	}
 
-	// Accept any prices that are within 2𝜎, or for which we couldn't get 𝜎
+	// accept any prices that are within 2𝜎, or for which we couldn't get 𝜎
 	for providerName, priceMap := range prices {
```

Comment:
nit: let's not shadow the `priceMap` variable here. It makes it hard to reason about what variable we're using.

Reply:
Updated @ 2c424a51a9165a335b09f926305e767b03bf9cc9